Add new crates `iommufd-bindings` and `iommufd-ioctls`

likebreath opened the pull request:
The `iommufd` workspace hosts libraries related to Rust bindings and wrappers to the [IOMMUFD](https://docs.kernel.org/userspace-api/iommufd.html) subsystem
from the Linux kernel. It currently consists of the following crates:

- `iommufd-bindings` -> Rust FFI bindings to IOMMUFD generated using bindgen
- `iommufd-ioctls` -> Safe wrappers over IOMMUFD uAPIs


These crates will be used by the `vfio-ioctls` crate and also Cloud Hypervisor to support the new vfio cdev mode [1].

[1] https://github.com/rust-vmm/vfio/issues/92

liuw commented on iommufd-ioctls/Cargo.toml:
```diff
@@ -0,0 +1,15 @@
+description = "Safe wrappers over IOMMUFD uAPIs"
+repository = "https://github.com/cloud-hypervisor/iommufd"
+readme = "README.md"
+edition = "2021"
```
Is there a reason to stay on the 2021 edition?

liuw reviewed (commented):
Ignoring the bindings, the code itself is just provides very simple wrappers.

It looks to me the `destroy` function is missing.

likebreath commented:
Thank you all for the fast turn around. I will address these comments later this week.

likebreath commented:
PTAL. All comments should be addressed. Main changes are:
1. Added Github actions as suggested above, workflow results: https://github.com/likebreath/iommufd/pull/1;
2. Added a basic unit tests to cover ioctls values;
3. Refined the SAFETY comments;

likebreath commented on .github/workflows/quality.yaml:
```diff
@@ -0,0 +1,39 @@
+        rust: [stable, beta]
+        target:
+          - aarch64-unknown-linux-gnu
+          - aarch64-unknown-linux-musl
```
Yes, that's actually the driven use case for us, e.g. Nvidia's Grace platforms.
